src/librustc/infer/error_reporting/anon_anon_conflict.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright 2012-2013 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Error Reporting for Anonymous Region Lifetime Errors
+//! where both the regions are anonymous.
+use hir;
+use infer::InferCtxt;
+use ty::{self, Region};
+use infer::region_inference::RegionResolutionError::*;
+use infer::region_inference::RegionResolutionError;
+use hir::map as hir_map;
+use middle::resolve_lifetime as rl;
+use hir::intravisit::{self, Visitor, NestedVisitorMap};
+
+// The visitor captures the corresponding `hir::Ty` of the
+// anonymous region.
+struct FindNestedTypeVisitor<'a, 'gcx: 'a + 'tcx, 'tcx: 'a> {
```

Comment:
e.g. "Consider a case where we have `fn foo(x: &mut Vec<&u8>, y: &u8) { x.push(y); }`. This would lead to a conflict between the two anonymous lifetimes for `&u8` in `x` and `y` respectively. This visitor would be invoked twice, once for each lifetime, and would walk the types like `&mut Vec<&u8>` and `&u8` looking for the HIR where that lifetime appears. This allows us to highlight the specific part of the type."